src/main/java/net/neoforged/neoforge/capabilities/CapabilityHooks.java

```diff
@@ -162,9 +162,9 @@ public static void invalidateCapsOnChunkUnload(ChunkEvent.Unload event) {
         }
     }
 
-    public static void cleanCapabilityListenerReferencesOnTick(TickEvent.LevelTickEvent event) {
-        if (event.phase == TickEvent.Phase.END && event.side.isServer()) {
-            ((ServerLevel) event.level).cleanCapabilityListenerReferences();
+    public static void cleanCapabilityListenerReferencesOnTick(LevelTickEvent.Post event) {
+        if (!event.getLevel().isClientSide()) {
```

Comment:
Would a `(event.getLevel() instance of ServerLevel serverLevel)` not be better here, removing the unchecked cast etc?